docs/getting-started/linting/FAQ.md

```diff
@@ -187,11 +187,24 @@ Instead, our tooling exists to **_augment_** TypeScript's built in checks with l
 
 The `no-undef` lint rule does not use TypeScript to determine the global variables that exist - instead, it relies upon ESLint's configuration.
 
-You can [manually define the set of allowed `globals` in your ESLint config](https://eslint.org/docs/user-guide/configuring/language-options#specifying-globals), and/or you can use one of the [pre-defined environment (`env`) configurations](https://eslint.org/docs/user-guide/configuring/language-options#specifying-environments).
+We strongly recommend that you do not use the `no-undef` lint rule on TypeScript projects. The checks it provides are already provided by TypeScript without the need for configuration - TypeScript just does this significantly better.
```

Comment:
IMHO TypeScript is doing a worse job than ESLint in multi-environment projects (e.g. browser, server, mocha) that have different globals. At least unless you are using project reference accurately.  Since project references don’t support glob patterns for configuration typescript cannot provide a "significantly better" check that works for all project structures. Especially not without the need of configuration, because per default typescript loads all global type definitions from `node_modules` even if they are not wanted.